src/data/items.h

```diff
@@ -1281,9 +1281,16 @@ const struct Item gItems[] =
     {
         .name = _("Escape Rope"),
         .itemId = ITEM_ESCAPE_ROPE,
-        .price = 550,
         .description = sEscapeRopeDesc,
+        #if I_KEY_ESCAPE_ROPE >= GEN_8
+        .price = 0,
+        .importance = 1,
+        .pocket = POCKET_KEY_ITEMS,
+        #else
+        .price = 550,
+        .importance = 0,
         .pocket = POCKET_ITEMS,
+        #endif
         .type = 2,
         .fieldUseFunc = ItemUseOutOfBattle_EscapeRope,
```

Comment:
BUG: 1 escape rope is still deducted when it's a key item:

![](https://i.imgur.com/jg5jNlx.gif)

Reply:
fixed it. in `item_use.c`:

```c
static void ItemUseOnFieldCB_EscapeRope(u8 taskId)
{
    Overworld_ResetStateAfterDigEscRope();
    #if I_KEY_ESCAPE_ROPE < GEN_8
      RemoveUsedItem();
    #endif
    gTasks[taskId].data[0] = 0;
    DisplayItemMessageOnField(taskId, gStringVar4, Task_UseDigEscapeRopeOnField);
}
```

Reply:
The include is already as part of "item.h", so it can already make use of the toggle.